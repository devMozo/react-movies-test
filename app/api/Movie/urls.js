import { MOVIES_API_KEY } from '../../constants/config';
import { QUERY_API_KEY } from '../../constants/queryStrings';
import { URL_MOVIE_API } from '../config';

export const URL_GET_MOVIE = movieId => `${URL_MOVIE_API}/movie/${movieId}?${QUERY_API_KEY}=${MOVIES_API_KEY}`;
