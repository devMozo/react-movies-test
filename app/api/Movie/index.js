import { URL_GET_MOVIE } from './urls';
import Axios from 'axios';

const getMovieById = movieId => {
    return Axios.get(URL_GET_MOVIE(movieId), {
        headers: {
            'Access-Control-Allow-Origin': '*',
        },
    })
        .then(response => {
            return response.data;
        })
        .catch(err => Promise.reject(err));
};

export { getMovieById };
