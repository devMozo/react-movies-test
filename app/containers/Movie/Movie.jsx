import React from 'react';
import { connect } from 'react-redux';
import { getMovie } from '../../redux-app/actions/MovieAction';
import FullMovie from '../../components/FullMovie/FullMovie';
import HandlerLoaders from '../../components/HandlerLoaders/HandlerLoaders';
import { addItem, removeItem } from '../../redux-app/actions/MyCartAction';
import './Movie.scss';

export class Movie extends React.PureComponent {
    getMovie = () => {
        const { getMovie, match } = this.props;
        getMovie(match.params.id);
    };

    onAdd = () => {
        const { movie, addItem } = this.props;
        addItem(movie);
    };

    onRemove = () => {
        const { movie, removeItem } = this.props;
        removeItem(movie.id);
    };

    isAdded = () => {
        const { movie, items } = this.props;
        return Boolean(items && items.find(item => item.id === movie.id));
    };

    componentDidMount() {
        this.getMovie();
    }

    render() {
        const { movie, loading, error } = this.props;

        return (
            <main className="Movie">
                <HandlerLoaders loading={loading} error={error} onTryAgain={this.getMovie}>
                    <FullMovie movie={movie} onAdd={this.onAdd} onRemove={this.onRemove} added={this.isAdded()} />
                </HandlerLoaders>
            </main>
        );
    }
}

const mapStateToProps = state => {
    return {
        ...state.movie,
        ...state.myCart,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        getMovie: movieId => {
            dispatch(getMovie(movieId));
        },
        addItem: item => {
            dispatch(addItem(item));
        },
        removeItem: idItem => {
            dispatch(removeItem(idItem));
        },
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Movie);
