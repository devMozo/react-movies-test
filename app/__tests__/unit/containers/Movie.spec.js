import React from 'react';
import { shallow, mount } from 'enzyme';
import { Movie } from '../../../containers/Movie/Movie';

const MOCK_UNIT_MOVIE = require('../../../__mocks__/unitMovie.json');

describe('<Movie />', () => {
    let PROPS;
    const mockedGetMovie = jest.fn();

    beforeEach(() => {
        PROPS = {
            movie: {},
            loading: false,
            error: false,
            getMovie: mockedGetMovie,
            match: {
                params: {
                    id: 320288,
                },
            },
        };
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Movie {...PROPS} />);
        expect(component).toMatchSnapshot();
    });

    it('should render all categories when the data was fetched', () => {
        PROPS = {
            ...PROPS,
            movie: MOCK_UNIT_MOVIE,
        };

        const component = mount(<Movie {...PROPS} />);

        expect(component.find('.FullMovie').exists()).toBeTruthy();
        expect(mockedGetMovie).toHaveBeenCalled();
    });
});
