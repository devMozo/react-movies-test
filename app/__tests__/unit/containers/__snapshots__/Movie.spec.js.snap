// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie /> should render correctly in "debug" mode 1`] = `
<main
  className="Movie"
>
  <Memo(HandlerLoaders)
    error={false}
    loading={false}
    onTryAgain={[Function]}
  >
    <Memo(FullMovie)
      added={false}
      movie={Object {}}
      onAdd={[Function]}
      onRemove={[Function]}
    />
  </Memo(HandlerLoaders)>
</main>
`;
